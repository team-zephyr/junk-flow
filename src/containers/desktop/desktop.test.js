// @jest
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { rendersCorrect, matchesSnapshot } from '../../utils/test-utilities';
import DesktopContainer, { Desktop } from './desktop';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';

const mockStore = configureMockStore([thunk]);
const store = mockStore();

const minProps = {
    contents: {},
    activePrograms: [],
    openProgram: () => { },
};

const getMinComponent = (otherProps = {}) => (
    <Desktop {...minProps} {...otherProps} />
);

const getMinConnectedComponent = (otherProps = {}) => (
    <DesktopContainer store={store} {...minProps} {...otherProps} />
);

it('Desktop renders correctly', () => {
    rendersCorrect(getMinComponent());
});

it('Desktop matches snapshot correctly', () => {
    matchesSnapshot(getMinComponent());
});

it('DesktopContainer renders correctly', () => {
    rendersCorrect(getMinConnectedComponent());
});

it('DesktopContainer matches snapshot', () => {
    matchesSnapshot(getMinConnectedComponent());
});