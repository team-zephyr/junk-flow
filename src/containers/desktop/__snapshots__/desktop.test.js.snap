// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Desktop matches snapshot correctly 1`] = `
<div>
  Im the Desktop
</div>
`;

exports[`DesktopContainer matches snapshot 1`] = `
<WrapperComponent
  activePrograms={Array []}
  closeProgramByProgramId={[Function]}
  closeProgramByWindowId={[Function]}
  contents={Object {}}
  openProgram={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
/>
`;
